src/Database/Driver/sparql/ConnectionInterface.php

```diff
@@ -34,6 +35,27 @@
    */
   public function query(string $query, array $args = [], array $options = []): Result;
 
+  /**
+   * Executes the actual query against the Sparql endpoint.
+   *
+   * @param string $query
+   *   The string query to execute.
+   * @param array $args
+   *   An array of arguments for the query.
+   * @param array $options
+   *   An associative array of options to control how the query is run.
+   *
+   * @return \EasyRdf\Sparql\Result
+   *   The query result.
+   *
+   * @throws \InvalidArgumentException
+   *   If $args value is passed but arguments replacement is not yet
+   *   supported. To be removed in #55.
+   *
+   * @see https://github.com/ec-europa/rdf_entity/issues/55
+   */
+  public function constructQuery(string $query, array $args = [], array $options = []): Graph;
```

Comment:
Looks like this method is doing exactly the same thing as `::query()`. Neither the code, nor the doc block is giving any explanation for what is needed or why we need an identical method. At least the documentation of this method should be improved to explain where is this used. Also, this PR seems to be required in https://github.com/ec-europa/joinup-dev/pull/1418. The PR is included there in composer.json but the method is never used. So, why we need this entire PR?

Reply:
The issue I'm trying to fix is in the return type hinting. Construct queries return Graph objects, while select queries return Result objects.
The only place where construct queries are used is here for the moment:
https://github.com/ec-europa/rdf_entity/pull/73/files#diff-3f45bd26a981f1ccd5a1675297de242eL52
This will solve the problem we have in prod at the moment:
e.g. https://joinup.ec.europa.eu/rdf-export/rdf_entity/http_e_f_fdata_ceuropa_ceu_fw21_fc911719d_b206e_b433f_b9639_b04c440cf7b26

Reply:
OK, but still needs a good comment/docs.